src/aggregator/aggregator/aggregator.go

```diff
@@ -740,8 +785,14 @@ func (agg *aggregator) tickInternal() {
 		tickResult           tickResult
 	)
 	for _, shard := range ownedShards {
-		shardTickResult := shard.Tick(perShardTickDuration)
-		tickResult = tickResult.merge(shardTickResult)
+		select {
+		case <-agg.doneCh:
```

Comment:
nit: we don't actually need to check here right? since we check within the tickShardFn when iterating over every metric right? just checking my understanding.

the only reason I mention this nit, is it at first it reads like we only check once per shard, which would be bad.

Reply:
The idea here is if we've signaled on doneCh, we can avoid ticking the remainder of the shards, I'll make this more explicit

Reply:
ah right. good point. nah it makes sense.